assayist/processor/base.py

```diff
@@ -290,6 +290,57 @@ def create_or_update_source_location(self, url, component, canonical_version=Non
         self.conditional_connect(sl.component, component)
         return sl.save()
 
+    def claim_file(self, base_dir, path_in_base_dir):
+        """
+        Claim (delete) a file in the base directory.
+
+        This method is used by analyzers to claim a file they've identified. All directories are
+        silently ignored.
+
+        :param str base_dir: the base directory to claim a file from
+        :param str path_in_container: the path to the file in the base directory to claim
+        :raises RuntimeError: when path_in_base_dir is the root directory or the path to the
+            base_dir is not a directory
+        """
+        if path_in_base_dir == '/':
+            raise RuntimeError('You cannot claim the root directory of a container')
+
+        file_path = path_in_base_dir.lstrip('/')
+
+        if not os.path.isdir(base_dir):
+            raise RuntimeError(f'The path "{base_dir}" is not a directory')
+
+        abs_base_dir = os.path.abspath(base_dir)
+
+        def _resolve_path(target):
+            """Resolve the first symbolic link in the path recursively."""
+            current_path = target
+            # Crawl upwards starting at the target until the base directory is reached
+            while current_path != abs_base_dir:
+                if os.path.islink(current_path):
+                    # Get the absolute path of the link's target but strip the starting slash
+                    link_target = os.path.abspath(os.readlink(current_path))[1:]
+                    # Find the path after the link, for instance, if the link is
+                    # `/opt/rh/httpd24/root/etc/httpd` => `/etc/httpd`, and the passed in target is
+                    # `/opt/rh/httpd24/root/etc/httpd/httpd.conf`, then we just want `httpd.conf`.
+                    path_after_link = os.path.relpath(target, current_path)
+                    # The resolved path for the above example would be the base directory plus
+                    # `etc/httpd/httpd.conf`
+                    resolved_path = os.path.join(abs_base_dir, link_target, path_after_link)
+                    # In the event there is more than one link in the path, call this closure
```

Comment:
"In case there is" would read better than "In the event there is" IMHO. We call the closure again without knowing whether there are more links.